include/sparrow/nullable.hpp

```diff
@@ -21,6 +21,16 @@
 
 #include "sparrow/mp_utils.hpp"
 
+// clang workaround: clang instantiates the constructor in SFINAE context,
+// which is incompatible with the implementation of standard libraries which
+// are not libc++.This leads to wrong compilation errors. Making the constructor
+// not constexpr prevents the compiler from instantiating it.
+#if defined(__clang__) && not defined(_LIBCPP_VERSION)
+#   define CONSTEXPR
```

Comment:
1. Please prefix this define to avoid name collisions.
2. Is this a general issue? if yes, move this somewhere more general. 
3. If this means issues with `constexpr` will appear somewhere else, maybe we need an issue about this and have a pass where we replace `constexpr` by a define.

Reply:
1. Sure, will do
2. Yes and no. I mean, the clang issue seems general, but it happens only with "incompatible" implementations of SFINAE.
3. So I don't know if we want to be conservative and use this macro only where we encounter the bug, or if we want to be extremely cautious and remove the constexpr everywhere when we build with clang and another standard library than libcxx.